tests/units/nulecule/test_nulecule.py

```diff
@@ -0,0 +1,97 @@
+import mock
+import unittest
+from atomicapp.nulecule.base import Nulecule
+
+
+class TestNuleculeRun(unittest.TestCase):
+    """Test Nulecule run"""
+
+    def test_run(self):
+        provider = 'docker'
+        dryrun = False
+        n = Nulecule('some-id', '0.0.2', {}, [], 'some/path')
+        mock_component_1 = mock.Mock()
+        mock_component_2 = mock.Mock()
+        n.components = [mock_component_1, mock_component_2]
+        n.run(provider)
+        mock_component_1.run.assert_called_once_with(provider, dryrun)
+        mock_component_2.run.assert_called_once_with(provider, dryrun)
+
+
+class TestNuleculeStop(unittest.TestCase):
+    """Test Nulecule stop"""
+
+    def test_stop(self):
+        provider = 'docker'
+        dryrun = False
+        n = Nulecule('some-id', '0.0.2', {}, [], 'some/path')
+        mock_component_1 = mock.Mock()
+        mock_component_2 = mock.Mock()
+        n.components = [mock_component_1, mock_component_2]
+        n.stop(provider)
+        mock_component_1.stop.assert_called_once_with(provider, dryrun)
+        mock_component_2.stop.assert_called_once_with(provider, dryrun)
+
+
+class TestNuleculeLoadConfig(unittest.TestCase):
+    """Test Nulecule load_config"""
+
+    def test_load_config(self):
+        config = {'group1': {'a': 'b'}}
+        n = Nulecule('some-id', '0.0.2', {}, [], 'some/path')
+        mock_component_1 = mock.Mock()
+        n.components = [mock_component_1]
+        mock_component_1.config = {
+            'group1': {'a': 'c', 'k': 'v'},
+            'group2': {'1': '2'}
+        }
+        n.load_config(config)
+        self.assertEqual(n.config, {
+            'group1': {'a': 'c', 'k': 'v'},
+            'group2': {'1': '2'}
+        })
+
+
+class TestNuleculeLoadComponents(unittest.TestCase):
+    """Test loading NuleculeComponents for a Nulecule"""
+
+    @mock.patch('atomicapp.nulecule.base.NuleculeComponent')
+    def test_load_components(self, MockNuleculeComponent):
+        graph = [
+            {
+                'name': 'app1',
+                'source': 'docker://somecontainer',
+                'params': []
+            },
+            {
+                'name': 'app2',
+                'artifacts': [
+                    {'a': 'b'}
+                ]
+            }
+        ]
+        n = Nulecule('some-id', '0.0.2', {}, graph, 'some/path')
+        n.load_components()
+        MockNuleculeComponent.assert_any_call(
+            graph[0]['name'], n.basepath, 'somecontainer',
+            graph[0]['params'], None)
+        MockNuleculeComponent.assert_any_call(
+            graph[1]['name'], n.basepath, None,
+            graph[1].get('params'), graph[1].get('artifacts'))
+
```

Comment:
can't use kwargs here if not used in the original function call. I will need to refactor the original function call as well, then.